include/IECore/IndexedIO.h

```diff
@@ -140,7 +141,8 @@ class IECORE_API IndexedIO : public RunTimeTyped
 		/// \param path A file or directory on disk. The appropriate reader for reading/writing is determined by the path's extension.
 		/// \param root The root point to 'mount' the structure.
 		/// \param mode A bitwise-ORed combination of constants which determine how the file system should be accessed.
-		static IndexedIOPtr create( const std::string &path, const EntryIDList &root, IndexedIO::OpenMode mode);
+		/// \param options A CompoundData of additional options to pass to the concrete IndexedIO instance create function.
+		static IndexedIOPtr create( const std::string &path, const EntryIDList &root, IndexedIO::OpenMode mode, const CompoundData *options = nullptr);
```

Comment:
It's always bugged me that `create()` _requires_ a filename, because there are perfectly valid implementations of IndexedIO that have nothing to do with files (MemoryIndexedIO for one, and fancy imaginary object stores for another). The same goes for SceneInterface, where we have LiveScenes where the filename is meaningless. I was hoping to remove the filename from `CreatorFn` and use a convention of a StringData "fileName" parameter instead. We could then have a pure `create()` method which didn't take a filename, and a convenience version that did take a filename, but just used it to add the right "fileName" parameter before calling the pure version. Any thoughts on that?

Reply:
> It's always bugged me that create() requires a filename, because there are perfectly valid implementations of IndexedIO that have nothing to do with files (MemoryIndexedIO for one, and fancy imaginary object stores for another). The same goes for SceneInterface, where we have LiveScenes where the filename is meaningless. I was hoping to remove the filename from CreatorFn and use a convention of a StringData "fileName" parameter instead. 

Sounds like a good idea.

> We could then have a pure create() method which didn't take a filename, and a convenience version that did take a filename, but just used it to add the right "fileName" parameter before calling the pure version. 

So would this use the Parameterised class? or Something else like my CompoundData options?

Reply:
> So would this use the Parameterised class? or Something else like my CompoundData options?

Something exactly like your CompoundData options. I was just thinking of naming the function argument `const CompoundData *parameters` rather than `const CompoundData *options` because that felt a bit more "primary" to me, rather than some additional stuff on top of the filename...

Reply:
Would you consider a follow up PR with this change?

Reply:
Yep, sure. Hopefully sooner rather than later though...